woocommerce/woocommerce-android
# Shipping labels: Edit address

Implements #3017.
|Editing|Country selector|Validation|
|-|-|-|
|![Screenshot_1608224838](https://user-images.githubusercontent.com/1522856/102535172-238c1b00-40a8-11eb-914b-8c7b1269466d.png) | ![Screenshot_1608224844](https://user-images.githubusercontent.com/1522856/102535553-a7460780-40a8-11eb-9a81-b1fb9ebe8511.png) | ![Screenshot_1608224849](https://user-images.githubusercontent.com/1522856/102535567-ad3be880-40a8-11eb-8921-36b6023e2bda.png) |

When the address validation fails, the edit address dialog is opened with an error message that indicates what the problem is. The edit address screen can also be opened by tapping on the Edit button in the flow-creation screen.

**To test:**
1. Go to an order that supports shipping label creation
2. Tap on the `Create` button
3. Notice the creation flow displays the flow-creation screen with the origin & destination address
4. Tap on the `Continue` button
5. Wait for the validation result
6. If the validation fails, the edit address screen is opened with an error message
7. If it passes, the next step in the creation flow is highlighted (shipping address)
8. In edit address screen, try to add an empty name, city, country or address
9. Tapping on the `Done` button validates the address and if it succeeds, the next step in the flow is shown
10. Tapping on the `Use address as entered` skips the validation

@Garance91540 I've modified the error banner a bit and included explicit buttons for viewing the address in Google Maps and contacting the customer (opens the dialer). Using links in the text so close to each other isn't great because of the reduced tap area. What do you think?
